policy/task/results.rego

```diff
@@ -0,0 +1,78 @@
+#
+# METADATA
+# title: Tekton Task result
+# description: Verify Tekton Task definitions provide expected results.
+#
+package policy.task.results
+
+import rego.v1
+
+import data.lib
+
+# METADATA
+# title: Required result defined
+# description: >-
+#   Verify if Task defines the required result. This is controlled by the `required_task_results`
+#   rule data key. By default this is empty making this rule a no-op.
+# custom:
+#   short_name: required
+#   failure_msg: '%s'
+#
+deny contains result if {
+	some err in errors
+	result := lib.result_helper(rego.metadata.chain(), [err])
+}
+
+# METADATA
+# title: Rule data provided
+# description: >-
+#   Confirm the expected `required_task_results` rule data key has been provided in the expected
+#   format.
+# custom:
+#   short_name: rule_data_provided
+#   failure_msg: '%s'
+#   solution: If provided, ensure the rule data is in the expected format.
+#   collections:
+#   - redhat
+#   - policy_data
+#
+deny contains result if {
+	some error in _rule_data_errors
+	result := lib.result_helper(rego.metadata.chain(), [error])
+}
+
+errors contains err if {
+	some required in lib.rule_data(_rule_data_key)
+	input.metadata.name == required.task
+	found := [result |
+		some result in input.spec.results
+		result.name == required.result
```

Comment:
Probably an unlikely edge case, but should we check that a value key is present as well?

Reply:
This is verifying a Task definition. There won't be a value. The result would be present in the status of a TaskRun.